libs/types/src/cdf/ballot-definition/convert.ts

```diff
@@ -0,0 +1,412 @@
+                '@type': 'BallotDefinition.CandidateContest',
+                '@id': contest.id,
+                ElectionDistrictId: contest.districtId,
+                Name: contest.title,
+                BallotTitle: text(contest.title),
+                VotesAllowed: contest.seats,
+                ContestOption: [
+                  ...contest.candidates.map(
+                    (candidate): Cdf.CandidateOption => ({
+                      '@type': 'BallotDefinition.CandidateOption',
+                      '@id': `option-${candidate.id}`,
+                      CandidateIds: [candidate.id],
+                      EndorsementPartyIds: candidate.partyIds,
+                    })
+                  ),
+                  // Create write-in options up to the number of votes allowed
+                  ...(contest.allowWriteIns
+                    ? take(contest.seats, naturals()).map(
+                        (writeInIndex): Cdf.CandidateOption => ({
+                          '@type': 'BallotDefinition.CandidateOption',
+                          '@id': `option-write-in-${writeInIndex}`,
+                          IsWriteIn: true,
+                        })
+                      )
+                    : []),
+                ],
+              };
+
+            case 'yesno':
+              return {
+                '@type': 'BallotDefinition.BallotMeasureContest',
+                '@id': contest.id,
+                ElectionDistrictId: contest.districtId,
+                Name: contest.title,
+                BallotTitle: text(contest.title),
+                FullText: text(contest.description),
+                ContestOption: [
+                  {
+                    '@type': 'BallotDefinition.BallotMeasureOption',
+                    '@id': contest.yesOption?.id ?? 'option-yes',
+                    Selection: text(contest.yesOption?.label ?? 'Yes'),
+                  },
+                  {
+                    '@type': 'BallotDefinition.BallotMeasureOption',
+                    '@id': contest.noOption?.id ?? 'option-no',
+                    Selection: text(contest.noOption?.label ?? 'No'),
+                  },
+                ],
+              };
+
+            /* istanbul ignore next */
+            default:
+              throwIllegalValue(contest);
+          }
+        }),
+
+        // TODO Eventually, we might want to list all of the contests included
+        // in the ballot style. Currently, our system infers it from the
+        // associated precincts' districts.
+        BallotStyle: vxfElection.ballotStyles.map(
+          (ballotStyle): Cdf.BallotStyle => ({
+            '@type': 'BallotDefinition.BallotStyle',
+            GpUnitIds: ballotStyle.precincts,
+            PartyIds: ballotStyle.partyId ? [ballotStyle.partyId] : undefined,
+            // In CDF, ballot styles don't have an id field. I think this might be
+            // because each ballot style can be uniquely identified by its
```

Comment:
This is fascinating, because the CVR CDF _does_ identify CVRs by a ballot style id.

Reply:
🤔 that is very odd